hudi-spark-datasource/hudi-spark3-common/src/main/scala/org/apache/spark/sql/HoodieSpark3CatalystPlanUtils.scala

```diff
@@ -52,8 +57,56 @@ abstract class HoodieSpark3CatalystPlanUtils extends HoodieCatalystPlansUtils {
     }
   }
 
-  override def toTableIdentifier(relation: UnresolvedRelation): TableIdentifier = {
-    relation.multipartIdentifier.asTableIdentifier
+  override def resolve(spark: SparkSession, relation: UnresolvedRelation): Option[CatalogTable] = {
+    val catalogManager = spark.sessionState.catalogManager
+    val nameParts = relation.multipartIdentifier
+    val expandedNameParts = expandIdentifier(spark, nameParts)
+    HoodieCatalogAndIdentifier.parse(catalogManager, expandedNameParts) match {
+      case Some((catalog, ident)) =>
+        CatalogV2Util.loadTable(catalog, ident) match {
+          case Some(table) =>
+            table match {
+              case v1Table: V1Table =>
+                Some(v1Table.v1Table)
+              case withFallback: V2TableWithV1Fallback =>
+                Some(withFallback.v1Table)
+              case _ =>
+                logWarning(s"It's not a hoodie table: $table")
+                None
+            }
+          case _ =>
+            logWarning(s"Can not load this catalog and identifier: ${catalog.name()}, $ident")
+            None
+        }
+      case _ =>
+        logWarning(s"Can not parse this name parts: ${expandedNameParts.mkString(",")}")
```

Comment:
ok.

Reply:
@alexeykudinkin if no other places need to be changed, can we just leave it?

Reply:
Why not fixing? The problem with existing code that you will get message like following which is very confusing:

```
Can not parse this name parts: a,b,c
```

It might momentarily seem like someone is using "," instead of "." in the name